### PR TITLE
Convert to .NET Core 1.0

### PR DESCRIPTION
**Pull request by scottmeyer**
I've modified this project so that I can use it in a .NET Core application. 

- All unit tests are passing but I can't run the integration tests. The client seems to be working for me in the sandbox environment.
- Upgraded NUnit to 3.4 for .NET Core support
- Unit tests run with `dotnet test`, however, you cannot exclude categories. Suggest moving the integration tests to a separate project.
- Change from `WebRequest` to `HttpClient` and `HttpRequestMessage`
- Hand code `SecurityElement.Escape` method as I could not find any compatible method in .NET Core

It looks like your build environment is Ruby/Mono and I don't have an environment like that to test with, so I have not modified the Rakefile.

### CODE REVIEW

**Comment by raymondberg** [comment]
Thanks for submitting this, @scottmeyer! It's going to take us a while to review this PR as it involves quite a few changes to the library. We really appreciate you diving in on this issue! We'll reach out with any questions or concerns as we review it.

**Comment by senwen** [comment]
Thanks @scottmeyer! Our company was about to embark on this.

**Comment by JoaquinCorcheroFMP** [comment]
Good morning,

We are looking as well to use a.net core version of your library, what is the state of the pull request? And when do you expect to have it fully reviewed?

Thank you

**Comment by abhurtun** [comment]
The pull request is out of date as the version of .net core is out of date we would like to submit a new pull request with the update .net core version.

**Comment by jrjrscottm** [comment]
@abhurtun are you saying the .net core version is out of date? It's RTM unless a newer version has been released that I haven't heard about.

I've applied the changes for 2.62.0, hope that helps.

File: global.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿{
+  "projects": [ "src", "test" ],
+  "sdk": {
+    "version": "1.0.0-preview2-003121"
```
Comment (jrjrscottm) [inline global.json]: @abhurtun It's preview2 here, not sure what you're saying?

Comment (jrjrscottm) [inline global.json]: I do see 1.0.0-rc1-final for `Microsoft.AspNet.WebUtilities` but that's the latest currently available.

Comment (senwen) [inline global.json]: @jrjrscottm Try updating to Microsoft.**AspNetCore**.WebUtilities.

Comment (scottmeyer) [inline global.json]: Noted and updated. Thanks

**Comment by abhurtun** [comment]
@scottmeyer you are correct i was looking at project.json and not global.json

**Comment by jamesford-fmp** [comment]
Quite keen to use this port. Is anyone reviewing it?

**Comment by abhurtun** [comment]
@scottmeyer @raymondberg Hi any ETAs on when this pull request might be merged in please?

**Comment by jdlc** [comment]
Hi everyone, thanks for your interest and comments on this PR. We are actively working on providing a .NET Core compatible client library, but we can't provide an ETA at this point. Thanks for your patience and enthusiasm!

**Comment by senwen** [comment]
@scottmeyer, 
We started integration of your fork and things run fine in Windows but received "System.PlatformNotSupportedException: Operation is not supported on this platform" on *nix and Mac. Looks like System.Net.Http.WinHttpHandler is being used. Seems like a more work is needed to make it truly cross-platform. 

@jdlc, 
Can we assume your team will make the official .NET core version truly cross-platform (*nix/pc/mac) compatible? Are you starting with this pull request? Trying to gauge how much effort we spend on this. 

Cheers.

**Comment by scottmeyer** [comment]
@senwen That will either need to be removed, or implemented by the owner. After reading this [blog post](https://www.braintreepayments.com/blog/when-rest-isnt-good-enough/) under the Security section, I thought it was going to be required.  It's there to validate the server certificate, but isn't actually validating anything yet (just returns true). 

However, I can't see anywhere in the existing code that the server certificate is being validated

**Comment by senwen** [comment]
Agreed @scottmeyer . For now, we commented it out until we here back from folks at Braintree.

**Comment by jdlc** [comment]
Closing this PR, we have released a version compatible with .NET Core. Thanks for your suggestion! https://github.com/braintree/braintree_dotnet/releases/tag/3.0.0

**Pull request closed without merging**